src/IW/Effects/Cabal.hs

```diff
@@ -0,0 +1,75 @@
+{- | This module contains the class definitiion of @MonadCabal@ and
+an instance of @MonadCabal@ for the @App@ monad. Instances of
+@MonadCabal@ have a @getCabalCategories@ action that returns @[Category]@
+given a @RepoOwner@ and @RepoName@. It does so by downloading a @.cabal@ file
+and parsing the @category@ field of the file.
+-}
+
+module IW.Effects.Cabal
+       ( getCabalCategories
+
+       -- * Internals
+       , getCabalCategoriesImpl
+       , repoCabalUrl
+       ) where
+
+import Data.Text (splitOn, strip)
+import Distribution.PackageDescription
+import Distribution.PackageDescription.Parsec (parseGenericPackageDescriptionMaybe)
+
+import IW.App (App)
+import IW.Core.Repo (RepoOwner (..), RepoName (..), Category (..))
+import IW.Core.Url (Url (..))
+import IW.Effects.Download (MonadDownload (..))
+
+
+-- | Describes a monad that returns @[Category]@ given a @RepoOwner@ and @RepoName@.
+class Monad m => MonadCabal m where
+    getCabalCategories :: RepoOwner -> RepoName -> m [Category]
+
+instance MonadCabal App where
+    getCabalCategories = getCabalCategoriesImpl
+
+type WithCabal env m = (MonadDownload m, WithLog env m)
+
+{- | This function may throw anyone of the errors inherited by the use of @downloadFile@
+defined in @IW.Effects.Download@. We are using @parseGenericPackageDescriptionMaybe@
+which will return @Nothing@ on an unsuccessful parse.
+-}
+getCabalCategoriesImpl :: WithCabal env m => RepoOwner -> RepoName -> m [Category]
+getCabalCategoriesImpl repoOwner repoName = do
+    cabalFile <- downloadFile $ repoCabalUrl repoOwner repoName
+    case parseGenericPackageDescriptionMaybe cabalFile of
+        Nothing -> do
+            log E $ "Couldn't parse file downloaded from " <> textUrl
+            pure []
+        Just genPkgDescr -> do
+            log I $ "Successfully parsed file downloaded from " <> textUrl
+            pure $ categoryNames genPkgDescr
+  where
+    textUrl :: Text
+    textUrl = unUrl $ repoCabalUrl repoOwner repoName
```

Comment:
`repoCabalUrl repoOwner repoName` duplicates twice: in `where` and inside the function.